autogen/agentchat/conversable_agent.py

```diff
@@ -611,25 +611,23 @@ def generate_code_execution_reply(
         if messages is None:
             messages = self._oai_messages[sender]
         last_n_messages = code_execution_config.pop("last_n_messages", 1)
+
+        # iterate through the last n messages reversly
+        # if code blocks are found, execute the code blocks and return the output
+        # if no code blocks are found, continue
         for i in range(min(len(messages), last_n_messages)):
             message = messages[-(i + 1)]
             code_blocks = extract_code(message["content"])
             if len(code_blocks) == 1 and code_blocks[0][0] == UNKNOWN:
-                # no code block is found, lang should be `UNKNOWN`
-
-                if i == last_n_messages - 1:
-                    code_execution_config["last_n_messages"] = last_n_messages
```

Comment:
this line shouldn't be removed.

Reply:
What's the logic for searching code block here

I assume it's searching the 'last_n_messages' from `code_execution_config` reversely and run the first code block that detected? In that case `last_n_messages` would be a read-only config and shouldn't need to be written back?

Reply:
The variable is popped out of the dict before the loop starts. So we need to restore it before returning.